tools/check-wininstaller-translations.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+set -eu
+
+BASE_DIR=src/res/translation/wininstaller/
+BASE_LANG=en
+INSTALLERLNG=installerlng.nsi
+BASE_LANGSTRINGS=$(grep LangString "${BASE_DIR}/${BASE_LANG}.nsi" | cut -d' ' -f2)
+EXIT=0
+for LANGUAGE_FILE in $(ls -1 src/res/translation/wininstaller/??.nsi | grep -vF "${BASE_LANG}.nsi"); do
+		echo
+		echo "* ${LANGUAGE_FILE}"
+		echo -n "  - Checking language file is included in ${INSTALLERLNG}... "
+		if grep -qF $(tr '/' '\\' <<<"${LANGUAGE_FILE}") "${BASE_DIR}/${INSTALLERLNG}"; then
+				echo "ok"
+		else
+				echo "ERROR"
+				EXIT=1
+		fi
+
+		echo -n "  - Checking language file for empty strings... "
+		if grep -qF '""' "${LANGUAGE_FILE}"; then
+				echo "ERROR, found empty string(s)"
+				EXIT=1
+		else
+				echo "ok"
+		fi
+
+		echo -n "  - Checking for comment leftovers from original file... "
+		if head -n1 "${LANGUAGE_FILE}" | grep -qF English; then
+				echo "ERROR, found 'English' in header"
+				EXIT=1
+		else
+				echo "ok"
+		fi
+
+		echo -n "  - Checking for wrong macros... "
+		LANG_MACROS="$(grep -oP '\$\{LANG_[^}]+\}' "${LANGUAGE_FILE}")"
+		if grep ENGLISH <<<"$LANG_MACROS"; then
+				echo "ERROR, found LANG_ENGLISH"
+				EXIT=1
+		else
+				echo "ok"
+		fi
+
+		echo -n "  - Checking for multiple lang macros... "
+		if [[ $(sort -u <<<"$LANG_MACROS" | wc -l) != 1 ]]; then
+				echo "ERROR, found multiple LANG_ macros"
+				EXIT=1
+		else
+				echo "ok"
+		fi
+
+		echo -n "  - Checking if LANG_ macro is in ${INSTALLERLNG}..."
+		if grep -qi '!insertmacro MUI_LANGUAGE "'$(sort -u <<<"${LANG_MACROS}" | sed -rne 's/\$\{LANG_(.*)\}/\1/p')'"' "$BASE_DIR/${INSTALLERLNG}"; then
```

Comment:
You could split then `; then` onto the next line - it looks nicer in bash, I think.

Reply:
> You could split then `; then` onto the next line - it looks nicer in bash, I think, and means the`;` is not required.

Hrm, I'd like to keep the `then` on the same line as other scripts do it the same way (e.g. `./linux/autorelease_linux.sh`, `./android/automated_build/install-qt.sh`). I think both styles are common though.

> (You could also split the long `$(...)` over a couple of lines with line breaks after and before the `(` and `)`, plus a break with `|\` before the `sed`. i.e. one command per line.)

I've moved the long call to a variable which makes the line shorter.